stackslib/src/chainstate/stacks/db/blocks.rs

```diff
@@ -193,10 +193,8 @@ impl BlockEventDispatcher for DummyEventDispatcher {
         _block_timestamp: Option<u64>,
         _coinbase_height: u64,
     ) {
-        assert!(
-            false,
-            "We should never try to announce to the dummy dispatcher"
-        );
+        error!("We should never try to announce to the dummy dispatcher");
+        panic!();
```

Comment:
Why not just:

```suggestion
        panic!("We should never try to announce to the dummy dispatcher");
```

Same for the one on line 209

Reply:
I have only received an okay to combine them in testing scenarios. The reason we cannot combine in non-test code is because the panic messages get lost sometimes. We have a custom panic handler which forces a core dump to happen on Unix-like systems. Until this lost message issue gets resolved, can't do :(

Reply:
Ok, that makes sense